onnxruntime/core/framework/kernel_registry_manager.cc

```diff
@@ -21,7 +21,6 @@ Status KernelRegistryManager::CreateKernel(const onnxruntime::Node& node,
   }
   Status status;
   {
-    std::lock_guard<OrtMutex> lock(lock_);
     for (auto& registry : custom_kernel_registries_) {
```

Comment:
Why is the lock unnecessary? Couldn't a concurrent call to InferenceSession::RegisterCustomRegistry change custom_kernel_registries_ at the same time?